node/pkg/watchers/aztec/block_fetcher.go

```diff
@@ -0,0 +1,144 @@
+package aztec
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// BlockFetcher defines the interface for retrieving Aztec chain data
+type BlockFetcher interface {
+	FetchPublicLogs(ctx context.Context, fromBlock, toBlock int) ([]ExtendedPublicLog, error)
+	FetchBlock(ctx context.Context, blockNumber int) (BlockInfo, error)
+}
+
+// aztecBlockFetcher is the implementation of BlockFetcher
+type aztecBlockFetcher struct {
+	rpcURL string
+	client HTTPClient
+	logger *zap.Logger
+}
+
+// NewAztecBlockFetcher creates a new block fetcher
+func NewAztecBlockFetcher(rpcURL string, client HTTPClient, logger *zap.Logger) BlockFetcher {
+	return &aztecBlockFetcher{
+		rpcURL: rpcURL,
+		client: client,
+		logger: logger,
+	}
+}
+
+// FetchPublicLogs gets logs for a specific block range
+func (f *aztecBlockFetcher) FetchPublicLogs(ctx context.Context, fromBlock, toBlock int) ([]ExtendedPublicLog, error) {
+	logFilter := map[string]any{
+		"fromBlock": fromBlock,
+		"toBlock":   toBlock,
+	}
+
+	payload := map[string]any{
+		"jsonrpc": "2.0",
+		"method":  "node_getPublicLogs",
+		"params":  []any{logFilter},
+		"id":      1,
```

Comment:
it will be worth documenting that guardians should set `ARCHIVER_MAX_LOGS` to a high number in the Aztec node since we don't support pagination in the client.